src/base/LruMap.h

```diff
@@ -148,39 +169,47 @@ LruMap<EntryValue, EntryCost>::add(const char *key, EntryValue *t)
     if (memLimit() == 0)
         return false;
 
-    index.push_front(new Entry(key, t));
+    auto e = new Entry(key, t, ttl);
+    memUsed_ += e->memCost() + e->key.length() + sizeof(Entry) + sizeof(MapPair);
```

Comment:
I still suspect that inaccurate overheads accounting is worse than no overheads accounting, but I am not going to press this issue if you are sure that inaccurate overheads accounting is better.

You may consider this particular problem addressed.